frog/imports/ui/GraphEditor/RemoteControllers/RemoteLibrary.jsx

```diff
@@ -0,0 +1,102 @@
+// @flow
+import React, { Component } from 'react';
+
+import { collectGraphs, importGraph } from '/imports/api/remoteGraphs';
+import { collectActivities } from '/imports/api/remoteActivities';
+import { addActivity } from '/imports/api/activities';
+import LibraryListComponent from './LibraryListComponent';
+
+const myFilter = (list: Array<any>, searchStr: string) =>
+  list
+    .filter(
+      x =>
+        (x.activity_type &&
+          x.activity_type.toLowerCase().includes(searchStr)) ||
+        x.title.toLowerCase().includes(searchStr) ||
+        x.description.toLowerCase().includes(searchStr) ||
+        x.tags.find(y => y.toLowerCase().includes(searchStr)) !== undefined
+    )
+    .sort((x: Object, y: Object) => (x.title < y.title ? -1 : 1));
+
+class Library extends Component<Object> {
+  componentWillMount() {
+    this.props.setImportList([]);
```

Comment:
Here it seems like you reset the import list every time you open the library panel - then why do you need to pass it in from the root component?